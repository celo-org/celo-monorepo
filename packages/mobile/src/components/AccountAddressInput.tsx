import TextInputWithButtons from '@celo/react-components/components/TextInputWithButtons'
import Touchable from '@celo/react-components/components/Touchable'
import QRCodeBorderlessIcon from '@celo/react-components/icons/QRCodeBorderless'
import colors from '@celo/react-components/styles/colors'
import React from 'react'
import { StyleSheet, TextInputProps, ViewStyle } from 'react-native'
import ClipboardAwarePasteIcon from 'src/components/ClipboardAwarePasteIcon'
import { navigate } from 'src/navigator/NavigationService'
import { Screens } from 'src/navigator/Screens'

interface Props {
  inputContainerStyle?: ViewStyle
  inputStyle?: TextInputProps['style']
  accountAddress: string
  onAddressChanged: (address: string) => void
  color?: string
}

export default function AccountAddressInput({
  inputContainerStyle,
  inputStyle,
  accountAddress,
  onAddressChanged,
  color = colors.goldUI,
}: Props) {
  const onPressQrCode = () => {
    navigate(Screens.WithdrawCeloQrScannerScreen, {
      onAddressScanned: onAddressChanged,
    })
  }

  return (
    <TextInputWithButtons
      style={inputContainerStyle}
      inputStyle={inputStyle}
      placeholder={'0x1234...5678'}
      placeholderTextColor={colors.gray3}
      onChangeText={onAddressChanged}
      value={accountAddress}
      testID={'AccountAddress'}
    >
      <ClipboardAwarePasteIcon
        style={styles.paste}
        onChangeText={onAddressChanged}
        color={color}
        value={accountAddress}
        testID={'PasteButton'}
      />
      <Touchable testID={'ScanQR'} borderless={true} onPress={onPressQrCode}>
        <QRCodeBorderlessIcon height={32} color={color} />
      </Touchable>
    </TextInputWithButtons>
  )
}

const styles = StyleSheet.create({
  paste: {
    marginRight: 8,
  },
})
