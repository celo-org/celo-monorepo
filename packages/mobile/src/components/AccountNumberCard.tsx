import Card from '@celo/react-components/components/Card'
import React from 'react'
import { StyleSheet, ViewStyle } from 'react-native'
import AccountNumber from 'src/components/AccountNumber'

interface Props {
  address: string
  style?: ViewStyle
}

export default function AccountNumberCard({ address, style }: Props) {
  return (
    <Card style={[styles.container, style]} rounded={true}>
      <AccountNumber address={address} touchDisabled={true} />
    </Card>
  )
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
  },
})
