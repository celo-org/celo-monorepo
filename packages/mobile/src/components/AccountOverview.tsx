import colors from '@celo/react-components/styles/colors'
import fontStyles, { estimateFontSize } from '@celo/react-components/styles/fonts'
import variables from '@celo/react-components/styles/variables'
import BigNumber from 'bignumber.js'
import * as React from 'react'
import { WithTranslation } from 'react-i18next'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import { connect } from 'react-redux'
import componentWithAnalytics from 'src/analytics/wrapper'
import CurrencyDisplay from 'src/components/CurrencyDisplay'
import Styles from 'src/components/Styles'
import { ExchangeRatePair } from 'src/exchange/reducer'
import { CURRENCIES, CURRENCY_ENUM } from 'src/geth/consts'
import { startBalanceAutorefresh, stopBalanceAutorefresh } from 'src/home/actions'
import { Namespaces, withTranslation } from 'src/i18n'
import { navigate } from 'src/navigator/NavigationService'
import { Screens } from 'src/navigator/Screens'
import { RootState } from 'src/redux/reducers'
import { getMoneyDisplayValue } from 'src/utils/formatting'

interface StateProps {
  exchangeRatePair: ExchangeRatePair | null
  goldEducationCompleted: boolean
  stableEducationCompleted: boolean
  goldBalance: string | null
  dollarBalance: string | null
}

interface OwnProps {
  testID: string
}

interface DispatchProps {
  startBalanceAutorefresh: typeof startBalanceAutorefresh
  stopBalanceAutorefresh: typeof stopBalanceAutorefresh
}

type Props = StateProps & DispatchProps & WithTranslation & OwnProps

const mapStateToProps = (state: RootState): StateProps => {
  return {
    exchangeRatePair: state.exchange.exchangeRatePair,
    // Disable education for now
    goldEducationCompleted: true, // state.goldToken.educationCompleted,
    stableEducationCompleted: true, // state.stableToken.educationCompleted,
    goldBalance: state.goldToken.balance,
    dollarBalance: state.stableToken.balance,
  }
}

export class AccountOverview extends React.Component<Props> {
  goToGoldTokenEducation = () => {
    navigate(Screens.GoldEducation)
  }

  goToStableTokenEducation = () => {
    navigate(Screens.DollarEducation)
  }

  getFontSize(balance: any, dot: boolean) {
    const numLength = getMoneyDisplayValue(balance).length
    const deviceWidth = variables.width
    const dotOffset = dot ? 15 : 0
    const containerWidth = deviceWidth / 2 - 60 - dotOffset // 60 = padding
    const fontSize = estimateFontSize(36, numLength, containerWidth)
    return fontSize
  }

  componentDidMount() {
    this.props.startBalanceAutorefresh()
  }

  componentWillUnmount() {
    this.props.stopBalanceAutorefresh()
  }

  render() {
    const { t, testID, goldBalance, dollarBalance } = this.props

    return (
      <View testID={testID}>
        <View>
          <View style={style.currencyContainer}>
            <View style={[style.currencyArea, Styles.center]} testID={`${testID}/dollarBalance`}>
              <Text style={[style.currencyLabel, fontStyles.bodySmall]}>
                {t('global:celoDollars') + ' ' + CURRENCIES[CURRENCY_ENUM.DOLLAR].code}
              </Text>
              <TouchableOpacity
                onPress={this.goToStableTokenEducation}
                disabled={this.props.stableEducationCompleted}
                style={[style.education, !this.props.stableEducationCompleted && style.dotOffset]}
              >
                <CurrencyDisplay
                  amount={new BigNumber(dollarBalance || 0)}
                  size={this.getFontSize(dollarBalance, !this.props.stableEducationCompleted)}
                  type={CURRENCY_ENUM.DOLLAR}
                />
              </TouchableOpacity>
            </View>
            <View style={style.line} />
            <View style={[style.currencyArea]} testID={`${testID}/goldBalance`}>
              <Text style={[style.currencyLabel, fontStyles.bodySmall]}>
                {t('global:celoGold') + ' ' + CURRENCIES[CURRENCY_ENUM.GOLD].code}
              </Text>
              <TouchableOpacity
                onPress={this.goToGoldTokenEducation}
                disabled={this.props.goldEducationCompleted}
                style={[style.education, !this.props.goldEducationCompleted && style.dotOffset]}
              >
                <CurrencyDisplay
                  amount={new BigNumber(goldBalance || 0)}
                  size={this.getFontSize(goldBalance, !this.props.goldEducationCompleted)}
                  type={CURRENCY_ENUM.GOLD}
                />
              </TouchableOpacity>
            </View>
          </View>
        </View>
      </View>
    )
  }
}

const style = StyleSheet.create({
  currencyContainer: {
    flexDirection: 'row',
    position: 'relative',
    paddingHorizontal: variables.contentPadding * 2,
    paddingTop: 15,
    paddingBottom: 5,
  },
  line: {
    alignSelf: 'center',
    height: 50,
    width: 1,
    marginTop: 10,
    backgroundColor: colors.darkLightest,
  },
  currencyArea: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  nativeBalanceRow: {
    flexDirection: 'row',
  },
  currencyLabel: {
    lineHeight: 30,
  },
  nativeBalance: {
    marginTop: -1,
    marginLeft: 3,
  },
  dollarBalance: {
    color: colors.celoGreen,
  },
  goldBalance: {
    color: colors.celoGold,
  },
  currency: {
    textAlignVertical: 'bottom',
  },
  progressBar: {
    position: 'relative',
  },
  education: {
    flexDirection: 'row',
    alignItems: 'center',
    overflow: 'visible',
  },
  dotOffset: {
    paddingLeft: 10,
  },
  dot: {
    padding: 10,
  },
})

export default componentWithAnalytics(
  connect<StateProps, DispatchProps, OwnProps, RootState>(
    mapStateToProps,
    { startBalanceAutorefresh, stopBalanceAutorefresh }
  )(withTranslation(Namespaces.walletFlow5)(AccountOverview))
)
