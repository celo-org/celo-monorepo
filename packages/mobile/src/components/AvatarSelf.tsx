import * as React from 'react'
import { useSelector } from 'react-redux'
import { e164NumberSelector, nameSelector, userContactDetailsSelector } from 'src/account/reducer'
import Avatar from 'src/components/Avatar'
import { RecipientKind, RecipientWithContact } from 'src/recipients/recipient'

// An avatar for the wallet user themselves
export function AvatarSelf() {
  const displayName = useSelector(nameSelector)
  const e164PhoneNumber = useSelector(e164NumberSelector)
  const contactDetails = useSelector(userContactDetailsSelector)

  // Recipient refering to the wallet user, used for the avatar
  const recipient: RecipientWithContact = {
    kind: RecipientKind.Contact,
    contactId: contactDetails.contactId || 'none',
    thumbnailPath: contactDetails.thumbnailPath || undefined,
    displayName,
    e164PhoneNumber,
  }

  return <Avatar recipient={recipient} />
}
