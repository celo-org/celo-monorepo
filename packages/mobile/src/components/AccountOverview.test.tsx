import * as React from 'react'
import 'react-native'
import * as renderer from 'react-test-renderer'
import { AccountOverview } from 'src/components/AccountOverview'
import { ExchangeRatePair } from 'src/exchange/reducer'
import { getMockI18nProps } from 'test/utils'

const SAMPLE_BALANCE = '55.00001'
const exchangeRatePair: ExchangeRatePair = { goldMaker: '0.11', dollarMaker: '10' }

it('renders correctly when ready', () => {
  const tree = renderer.create(
    <AccountOverview
      dollarBalance={SAMPLE_BALANCE}
      goldBalance={SAMPLE_BALANCE}
      exchangeRatePair={exchangeRatePair}
      goldEducationCompleted={true}
      stableEducationCompleted={true}
      testID={'SnapshotAccountOverview'}
      startBalanceAutorefresh={jest.fn()}
      stopBalanceAutorefresh={jest.fn()}
      {...getMockI18nProps()}
    />
  )
  expect(tree).toMatchSnapshot()
})

it('renders correctly when not ready', () => {
  const tree = renderer.create(
    <AccountOverview
      dollarBalance={SAMPLE_BALANCE}
      goldBalance={SAMPLE_BALANCE}
      exchangeRatePair={exchangeRatePair}
      goldEducationCompleted={true}
      stableEducationCompleted={true}
      testID={'SnapshotAccountOverview'}
      startBalanceAutorefresh={jest.fn()}
      stopBalanceAutorefresh={jest.fn()}
      {...getMockI18nProps()}
    />
  )
  expect(tree).toMatchSnapshot()
})

it('renders correctly when transaction pending', () => {
  const tree = renderer.create(
    <AccountOverview
      dollarBalance={SAMPLE_BALANCE}
      goldBalance={SAMPLE_BALANCE}
      exchangeRatePair={exchangeRatePair}
      goldEducationCompleted={true}
      stableEducationCompleted={true}
      testID={'SnapshotAccountOverview'}
      startBalanceAutorefresh={jest.fn()}
      stopBalanceAutorefresh={jest.fn()}
      {...getMockI18nProps()}
    />
  )
  expect(tree).toMatchSnapshot()
})

it("renders correctly when Gold education NUX flow hasn't been completed", () => {
  const tree = renderer.create(
    <AccountOverview
      dollarBalance={SAMPLE_BALANCE}
      goldBalance={SAMPLE_BALANCE}
      exchangeRatePair={exchangeRatePair}
      goldEducationCompleted={false}
      stableEducationCompleted={true}
      testID={'SnapshotAccountOverview'}
      startBalanceAutorefresh={jest.fn()}
      stopBalanceAutorefresh={jest.fn()}
      {...getMockI18nProps()}
    />
  )
  expect(tree).toMatchSnapshot()
})

it("renders correctly when Dollar education NUX flow hasn't been completed", () => {
  const tree = renderer.create(
    <AccountOverview
      dollarBalance={SAMPLE_BALANCE}
      goldBalance={SAMPLE_BALANCE}
      exchangeRatePair={exchangeRatePair}
      goldEducationCompleted={true}
      stableEducationCompleted={false}
      testID={'SnapshotAccountOverview'}
      startBalanceAutorefresh={jest.fn()}
      stopBalanceAutorefresh={jest.fn()}
      {...getMockI18nProps()}
    />
  )
  expect(tree).toMatchSnapshot()
})
