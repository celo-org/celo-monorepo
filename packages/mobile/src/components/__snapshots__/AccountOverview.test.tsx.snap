// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when Dollar education NUX flow hasn't been completed 1`] = `
<View
  testID="SnapshotAccountOverview"
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingHorizontal": 32,
          "paddingTop": 15,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
        testID="SnapshotAccountOverview/dollarBalance"
      >
        <Text
          style={
            Array [
              Object {
                "lineHeight": 30,
              },
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Regular",
                "fontSize": 14,
                "lineHeight": 18,
              },
            ]
          }
        >
          global:celoDollars cUSD
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "opacity": 1,
              "overflow": "visible",
              "paddingLeft": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "marginTop": 5,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#42D689",
                    "fontSize": 21,
                    "lineHeight": 29,
                    "transform": Array [
                      Object {
                        "translateY": 2,
                      },
                    ],
                  },
                ]
              }
            >
              $
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "paddingHorizontal": 3,
                  },
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#42D689",
                    "fontSize": 36,
                    "lineHeight": 47,
                  },
                ]
              }
            >
              55.00
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#EEEEEE",
            "height": 50,
            "marginTop": 10,
            "width": 1,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
        testID="SnapshotAccountOverview/goldBalance"
      >
        <Text
          style={
            Array [
              Object {
                "lineHeight": 30,
              },
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Regular",
                "fontSize": 14,
                "lineHeight": 18,
              },
            ]
          }
        >
          global:celoGold cGLD
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "opacity": 1,
              "overflow": "visible",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "marginTop": 5,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#FBCC5C",
                    "fontSize": 21,
                    "lineHeight": 29,
                    "transform": Array [
                      Object {
                        "translateY": 2,
                      },
                    ],
                  },
                ]
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "paddingHorizontal": 3,
                  },
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#FBCC5C",
                    "fontSize": 36,
                    "lineHeight": 47,
                  },
                ]
              }
            >
              55.00
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly when Gold education NUX flow hasn't been completed 1`] = `
<View
  testID="SnapshotAccountOverview"
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingHorizontal": 32,
          "paddingTop": 15,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
        testID="SnapshotAccountOverview/dollarBalance"
      >
        <Text
          style={
            Array [
              Object {
                "lineHeight": 30,
              },
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Regular",
                "fontSize": 14,
                "lineHeight": 18,
              },
            ]
          }
        >
          global:celoDollars cUSD
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "opacity": 1,
              "overflow": "visible",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "marginTop": 5,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#42D689",
                    "fontSize": 21,
                    "lineHeight": 29,
                    "transform": Array [
                      Object {
                        "translateY": 2,
                      },
                    ],
                  },
                ]
              }
            >
              $
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "paddingHorizontal": 3,
                  },
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#42D689",
                    "fontSize": 36,
                    "lineHeight": 47,
                  },
                ]
              }
            >
              55.00
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#EEEEEE",
            "height": 50,
            "marginTop": 10,
            "width": 1,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
        testID="SnapshotAccountOverview/goldBalance"
      >
        <Text
          style={
            Array [
              Object {
                "lineHeight": 30,
              },
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Regular",
                "fontSize": 14,
                "lineHeight": 18,
              },
            ]
          }
        >
          global:celoGold cGLD
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "opacity": 1,
              "overflow": "visible",
              "paddingLeft": 10,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "marginTop": 5,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#FBCC5C",
                    "fontSize": 21,
                    "lineHeight": 29,
                    "transform": Array [
                      Object {
                        "translateY": 2,
                      },
                    ],
                  },
                ]
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "paddingHorizontal": 3,
                  },
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#FBCC5C",
                    "fontSize": 36,
                    "lineHeight": 47,
                  },
                ]
              }
            >
              55.00
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly when not ready 1`] = `
<View
  testID="SnapshotAccountOverview"
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingHorizontal": 32,
          "paddingTop": 15,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
        testID="SnapshotAccountOverview/dollarBalance"
      >
        <Text
          style={
            Array [
              Object {
                "lineHeight": 30,
              },
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Regular",
                "fontSize": 14,
                "lineHeight": 18,
              },
            ]
          }
        >
          global:celoDollars cUSD
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "opacity": 1,
              "overflow": "visible",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "marginTop": 5,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#42D689",
                    "fontSize": 21,
                    "lineHeight": 29,
                    "transform": Array [
                      Object {
                        "translateY": 2,
                      },
                    ],
                  },
                ]
              }
            >
              $
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "paddingHorizontal": 3,
                  },
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#42D689",
                    "fontSize": 36,
                    "lineHeight": 47,
                  },
                ]
              }
            >
              55.00
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#EEEEEE",
            "height": 50,
            "marginTop": 10,
            "width": 1,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
        testID="SnapshotAccountOverview/goldBalance"
      >
        <Text
          style={
            Array [
              Object {
                "lineHeight": 30,
              },
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Regular",
                "fontSize": 14,
                "lineHeight": 18,
              },
            ]
          }
        >
          global:celoGold cGLD
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "opacity": 1,
              "overflow": "visible",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "marginTop": 5,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#FBCC5C",
                    "fontSize": 21,
                    "lineHeight": 29,
                    "transform": Array [
                      Object {
                        "translateY": 2,
                      },
                    ],
                  },
                ]
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "paddingHorizontal": 3,
                  },
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#FBCC5C",
                    "fontSize": 36,
                    "lineHeight": 47,
                  },
                ]
              }
            >
              55.00
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly when ready 1`] = `
<View
  testID="SnapshotAccountOverview"
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingHorizontal": 32,
          "paddingTop": 15,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
        testID="SnapshotAccountOverview/dollarBalance"
      >
        <Text
          style={
            Array [
              Object {
                "lineHeight": 30,
              },
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Regular",
                "fontSize": 14,
                "lineHeight": 18,
              },
            ]
          }
        >
          global:celoDollars cUSD
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "opacity": 1,
              "overflow": "visible",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "marginTop": 5,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#42D689",
                    "fontSize": 21,
                    "lineHeight": 29,
                    "transform": Array [
                      Object {
                        "translateY": 2,
                      },
                    ],
                  },
                ]
              }
            >
              $
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "paddingHorizontal": 3,
                  },
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#42D689",
                    "fontSize": 36,
                    "lineHeight": 47,
                  },
                ]
              }
            >
              55.00
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#EEEEEE",
            "height": 50,
            "marginTop": 10,
            "width": 1,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
        testID="SnapshotAccountOverview/goldBalance"
      >
        <Text
          style={
            Array [
              Object {
                "lineHeight": 30,
              },
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Regular",
                "fontSize": 14,
                "lineHeight": 18,
              },
            ]
          }
        >
          global:celoGold cGLD
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "opacity": 1,
              "overflow": "visible",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "marginTop": 5,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#FBCC5C",
                    "fontSize": 21,
                    "lineHeight": 29,
                    "transform": Array [
                      Object {
                        "translateY": 2,
                      },
                    ],
                  },
                ]
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "paddingHorizontal": 3,
                  },
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#FBCC5C",
                    "fontSize": 36,
                    "lineHeight": 47,
                  },
                ]
              }
            >
              55.00
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`renders correctly when transaction pending 1`] = `
<View
  testID="SnapshotAccountOverview"
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingBottom": 5,
          "paddingHorizontal": 32,
          "paddingTop": 15,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
        testID="SnapshotAccountOverview/dollarBalance"
      >
        <Text
          style={
            Array [
              Object {
                "lineHeight": 30,
              },
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Regular",
                "fontSize": 14,
                "lineHeight": 18,
              },
            ]
          }
        >
          global:celoDollars cUSD
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "opacity": 1,
              "overflow": "visible",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "marginTop": 5,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#42D689",
                    "fontSize": 21,
                    "lineHeight": 29,
                    "transform": Array [
                      Object {
                        "translateY": 2,
                      },
                    ],
                  },
                ]
              }
            >
              $
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "paddingHorizontal": 3,
                  },
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#42D689",
                    "fontSize": 36,
                    "lineHeight": 47,
                  },
                ]
              }
            >
              55.00
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#EEEEEE",
            "height": 50,
            "marginTop": 10,
            "width": 1,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            },
          ]
        }
        testID="SnapshotAccountOverview/goldBalance"
      >
        <Text
          style={
            Array [
              Object {
                "lineHeight": 30,
              },
              Object {
                "color": "#2E3338",
                "fontFamily": "Hind-Regular",
                "fontSize": 14,
                "lineHeight": 18,
              },
            ]
          }
        >
          global:celoGold cGLD
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "opacity": 1,
              "overflow": "visible",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flexDirection": "row",
                "marginTop": 5,
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#FBCC5C",
                    "fontSize": 21,
                    "lineHeight": 29,
                    "transform": Array [
                      Object {
                        "translateY": 2,
                      },
                    ],
                  },
                ]
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "paddingHorizontal": 3,
                  },
                  Object {
                    "fontFamily": "Hind-Regular",
                  },
                  Object {
                    "color": "#FBCC5C",
                    "fontSize": 36,
                    "lineHeight": 47,
                  },
                ]
              }
            >
              55.00
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
